misc/semver-checker/modules/semver-checker-core/src/main/java/io/ballerina/semver/checker/comparator/ClassComparator.java

```diff
@@ -0,0 +1,296 @@
+        resourceDiffExtractor.getAdditions().forEach((name, function) -> {
+            FunctionDiff.Builder funcDiffBuilder = new FunctionDiff.Builder(function, null);
+            funcDiffBuilder.withVersionImpact(SemverImpact.MINOR).build().ifPresent(memberDiffs::add);
+        });
+        resourceDiffExtractor.getRemovals().forEach((name, function) -> {
+            FunctionDiff.Builder funcDiffBuilder = new FunctionDiff.Builder(null, function);
+            funcDiffBuilder.withVersionImpact(SemverImpact.MAJOR).build().ifPresent(memberDiffs::add);
+        });
+        resourceDiffExtractor.getCommons().forEach((name, functions) -> new FunctionComparator(functions.getKey(),
+                functions.getValue()).computeDiff().ifPresent(memberDiffs::add));
+
+        DiffExtractor<ObjectFieldNode> varDiffExtractor = new DiffExtractor<>(newClassFields, oldClassFields);
+        varDiffExtractor.getAdditions().forEach((name, field) -> {
+            NodeDiffBuilder classVarDiffBuilder = new ObjectFieldDiff.Builder(field, null);
+            classVarDiffBuilder.withVersionImpact(SemverImpact.MINOR).withKind(DiffKind.OBJECT_FIELD).build()
+                    .ifPresent(memberDiffs::add);
+        });
+        varDiffExtractor.getRemovals().forEach((name, field) -> {
+            NodeDiffBuilder classVarDiffBuilder = new ObjectFieldDiff.Builder(null, field);
+            classVarDiffBuilder.withVersionImpact(SemverImpact.MAJOR).withKind(DiffKind.OBJECT_FIELD).build()
+                    .ifPresent(memberDiffs::add);
+        });
+        varDiffExtractor.getCommons().forEach((name, classFields) -> new ObjectFieldComparator(classFields.getKey(),
+                classFields.getValue()).computeDiff().ifPresent(memberDiffs::add));
+
+        return memberDiffs;
+    }
+
+    private void extractClassMembers(ClassDefinitionNode clazz, boolean isNewClass) {
+        clazz.members().forEach(member -> {
+            switch (member.kind()) {
+                case OBJECT_METHOD_DEFINITION:
+                case RESOURCE_ACCESSOR_DEFINITION:
+                    FunctionDefinitionNode funcNode = (FunctionDefinitionNode) member;
+                    if (isNewClass) {
+                        newFunctions.put(getFunctionIdentifier(funcNode), funcNode);
+                    } else {
+                        oldFunctions.put(getFunctionIdentifier(funcNode), funcNode);
+                    }
+                    break;
+                case OBJECT_FIELD:
+                    ObjectFieldNode objectField = (ObjectFieldNode) member;
+                    if (isNewClass) {
+                        newClassFields.put(objectField.fieldName().text().trim(), objectField);
+                    } else {
+                        oldClassFields.put(objectField.fieldName().text().trim(), objectField);
+                    }
+                    break;
+                default:
+                    // Todo: prompt a warning on the unsupported class member types
```

Comment:
Are we going to provide the warning within this fix?

Reply:
Since we still haven't added a diagnostic collector design, we dont have a way to collect the warning here and proceed. Thats why I've added the todo item so that we can revisit the diagnostics handling logic later.

Reply:
That's wont be straightforward since currently we don't have access to the PrintStream within the comparator tree model. Also this default block is (almost) unreachable since we've already handled all possible cases. WDYT?

Reply:
Sorry I  thought that we could access the PrintStream. Yeah, that is fine to handle it separately :)